modules/core/02-client/keeper/client.go

```diff
@@ -178,8 +196,12 @@ func (k Keeper) CheckMisbehaviourAndUpdateState(ctx sdk.Context, misbehaviour ex
 		return sdkerrors.Wrapf(types.ErrClientNotFound, "cannot check misbehaviour for client with ID %s", misbehaviour.GetClientID())
 	}
 
-	if clientState.IsFrozen() && clientState.GetFrozenHeight().LTE(misbehaviour.GetHeight()) {
-		return sdkerrors.Wrapf(types.ErrInvalidMisbehaviour, "client is already frozen at height ≤ misbehaviour height (%s ≤ %s)", clientState.GetFrozenHeight(), misbehaviour.GetHeight())
+	if clientState.IsFrozen() {
+		return sdkerrors.Wrap(types.ErrInvalidMisbehaviour, "client is already frozen")
+	}
+
+	if err := misbehaviour.ValidateBasic(); err != nil {
```

Comment:
My concern is if misbehaviour is passed in from another module rather than through the SDK msg that we designed

Reply:
Out of scope for this PR, let's open separate issue to discuss if there's disagreement

Reply:
> My concern is if misbehaviour is passed in from another module rather than through the SDK msg that we designed

This can be applied to all our handlers. Is there a use case in mind? We could turn these functions into message servers such that we aren't exposing functions that don't take messages

Reply:
Suppose with Interchain Staking I get a packet of Evidence which is chain-wide misbehaviour. I could freeze the underlying client during processing

Reply:
see #150